nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -412,23 +412,15 @@ def __init__(self, tensor_collectors: List[TensorCollector]) -> None:
 ##################################################
 
 
-class NoopReducer(TensorReducerBase):
+class RawReducer(TensorReducerBase):
     def __init__(self):
         super().__init__(inplace=False)
 
     def get_inplace_fn(self) -> Optional[InplaceInsertionFNType]:
         return None
 
-    def _reduce_out_of_place(self, x: List[TensorType]) -> List[TensorType]:
-        return x
-
-
-class RawReducer(NoopReducer):
-    def __init__(self):
-        super().__init__()
-
-    def __call__(self, x: List[Tensor]):
-        return self._reduce_out_of_place(x)
+    def _reduce_out_of_place(self, x: List[Tensor]) -> List[Tensor]:
+        return [tensor.clone() for tensor in x]
```

Comment:
What is the reason for `clone`? What about increased memory consumption here?
Also, the return format was changed (at least in terms of the final dimension). Was it tested for every use case in NNCF?

Reply:
during the work of reducer the tensor is reduced and used but when it comes to BC the value of the tensor from time to time looks like a garbage. After adding clone the value is stable so I do not know the exact reason but it looks like the without clone the initial tensor can be screwed up.
I tested on performance jobs and it passes but it could potentially lead to increase of memory consumption 
Why does return format is changed? I did not get it

Reply:
From a high perspective, it looks like we are hiding the bug since we do not know the exact reason for the `garbaged tensor` but are trying to avoid it by cloning. Is it eligible for ONNX only or also for OV/PT backends?
Also, unnecessary memory consumption may lead to huge regression in other backends and huge models in every scenario, since RawReducer is used in different algorithms.

The format wasn't changed, you're correct.

Reply:
My assumption that this tensor is cleaned by garbage collector as noone changes nncf.Tensor inside this reducer (this is proved because the issue is fixed by clone). 
I think clone is the right behavior here as we can not rely on the fact that the tensor came will not be modified/removed

Reply:
Any observations, on why the `numpy` version downgrade also resolves the issue?
From my side, these changes introduce regressions and look unnecessary because the initial reason for these is not clear.

Reply:
I searched for the differences in memory management for numpy between versions and didn't find any info.
What are exact regressions? I do not see any regression for now (all tests passed, jobs passed)

Reply:
Memory consumption changes weren't analysed as I understood. This is a potential regression.

Reply:
Why do you think so? I run 2 performance jobs with memory consumption analyze....

Reply:
Provide the memory analysis then.

Reply:
https://github.com/openvinotoolkit/nncf/pull/3081#issuecomment-2476131286

Reply:
Ok. I'll do it for you.
Reference (post_training_quantization_performance - 58):
![image](https://github.com/user-attachments/assets/40b8dee6-e70f-45cb-bd25-2556e3c4fa7c)
Your run:
![image](https://github.com/user-attachments/assets/9f66dac7-bee9-49fc-a40c-021a039053be)
This is only one example.

Reply:
good job, did you find anything suspicious, because I did not

Reply:
Memory consumption increased as we can see. Need to verify other models and experiment with some LLMs, since their consumer memory a way higher.
It also was decreased for some reason. This may be because of the test instability as well.

Reply:
I couldn't find any memory threshold for the tests in the code. Can you help here? What is the threshold?

Reply:
There are examples that are influenced by these changes have the same numbers:
develop:
![image](https://github.com/user-attachments/assets/96c0313d-3cae-429a-8ea6-290883cf258a)
PR:
![image](https://github.com/user-attachments/assets/8dd488d9-1dc9-4c80-bfef-512b1d9fde44)

develop:
![image](https://github.com/user-attachments/assets/82eadcb7-df46-45ab-a515-16d396e8909a)
PR:
![image](https://github.com/user-attachments/assets/1d941a9f-e259-4918-846b-bcda41fb4ce6)

Moreover, I looked at some nightly build and saw around 3% deviations there. 
Also I looked at LLM memory consumption tests as the most memory intensive and didn't see any difference. (manual job vs nightly)

Based on this data I do not see any regression here

Reply:
Please provide the LLM memory consumption graphs also.
As I found, here is your PR:
![image](https://github.com/user-attachments/assets/f270bff9-8395-4f5f-be1a-8ffd234c462f)
Latest develop run:
![image](https://github.com/user-attachments/assets/15dabc55-b7c2-422b-b242-a83a65cc36cc)

Reply:
You looked at data-free method which should not be influenced by the proposed changes

Reply:
As you wish.
I will not approve this change, since the issue analysis was not done properly.